### DISCUSSION TITLE
[BE] 회원 정보의 기본 이미지 저장 방식

### DISCUSSION THREAD

**Discussion by HyeonbinSa** [woowacourse-teams/2022-ternoko#189] [Ideas]
지난 스프린트 3에서 코치 정보 수정시 이미지 아이콘만 나오고 데이터는 Null로 저장된 적이 있습니다. 그래서 이런 문제를 방지하고자 이미지에 기본값을 넣자는 이야기가 나와 시도 중입니다. 

자세한 내용은 [노션](https://www.notion.so/NOT-NULL-vs-DEFAULT-5408d8a5072a4e1687901c9b639ad0b2)에 작성해보았는데 한번 읽어보시고 의견 부탁드리겠습니다. (주내용만 보고싶으시다면 하단의 시도 1, 시도 2를 읽어보시면 좋을 것 같습니다.)

이번 디스커션에서 기본 이미지 저장 방식에 대해 논의하고자 하는데요.
생각해본 방식으로는 
1. JPA에서 이미지 필드에 기본값을 주어 null일 경우 이를 사용하도록 하는 방식입니다.
    - 현재 방식에서 @DynamicInsert 와 같은 Annotation을 사용하는데 적용하기 위해서 추가 학습이 필요해보입니다.
2. 각 사용하는 위치(회원 가입, 정보 수정)에서 이미지가 null인지 검증하는 로직을 추가한다.
3. 기존 방식에서 DTO 필드의 오타로 인해 발생한 문제였는데 이를 해결하고 현재 상황을 보니 이미지가 null로 들어올 가능성이 없다는 생각이 들었습니다. 그래서 기존 방식을 유지하는 방식으로 가되 imageUrl 컬럼을 not null로 변경하여 null값이 들어올 때 예외를 던져주게끔만(백엔드 확인용)해도 괜찮겠다 생각이 들었습니다.

바쁘시겠지만 한번 읽어보시고 개인적으로 어떻게 하면 좋을지 의견 한번씩 부탁드리겠습니다.

**Comment by dongho108** [comment]
not null만 있어도 괜찮겠다는 생각이 들어요!
not null + default, (적용시 dynamic insert가 필요하다 하셨던) 적용하시겠다면 말리진 않겠습니다

**Reply by HyeonbinSa** [reply]
NOT NULL이 저도 좋을거같네요 ㅎㅎ;;;

**Comment by soominsohn** [comment]
3번 not null로 변경이 괜찮아 보입니다! 조금 더 예외케이스를 정확히 잡고 싶으면 3+2도 괜찮을 것 같네요.
1번 방법또한 학습해보면 좋을 키워드 같습니다!
하지만 현재상황에서는 3번에 한표 올리겠습니다 😄

**Reply by HyeonbinSa** [reply]
우선은 3번으로 진행하고 1번은 같이 학습하면서 진행해보면 좋을거같습니다. ㅎㅎ 의견 감사합니다 앤지

**Comment by her0807** [comment]
not null 한표 ! ☺️
